Pomodoro.Api/Controllers/SettingsController.cs

```diff
@@ -0,0 +1,141 @@
+        [SwaggerResponse(409, "Pomodoro settings for the current user already exist.")]
+        [SwaggerResponse(500, "An unhandled exception occurred on the server while executing the request.")]
+        public async Task<ActionResult<SettingsViewModel>> CreateSettings([FromBody] SettingsViewModel settingsVewModel)
+        {
+            if (this.UserId != settingsVewModel.UserId)
+            {
+                return this.BadRequest("UserId mismatch.");
+            }
+
+            var userSettings = await this.settingsService
+                .GetUserSettingsAsync(this.UserId);
+            if (userSettings != null)
+            {
+                return this.Conflict("Pomodoro settings for the current user already exist.");
+            }
+
+            var settingsModel = this.mapper.Map<SettingsModel>(settingsVewModel);
+            settingsModel = await this.settingsService.CreateSettingsAsync(settingsModel);
+
+            var createdViewModel = this.mapper.Map<SettingsViewModel>(settingsModel);
+            return this.CreatedAtAction(nameof(this.GetUserSettings), createdViewModel);
+        }
+
+        /// <summary>
+        /// Updates pomodoro settings related to the current user.
+        /// </summary>
+        /// <param name="id">Id of the pomodoro settings.</param>
+        /// <param name="settingsViewModel">Represents object to be updated.</param>
+        /// <returns>Updated object represented by <see cref="SettingsViewModel"/>.</returns>
+        [HttpPut("{id}")]
+        [ProducesResponseType(StatusCodes.Status202Accepted)]
+        [ProducesResponseType(StatusCodes.Status400BadRequest)]
+        [ProducesResponseType(StatusCodes.Status401Unauthorized)]
+        [ProducesResponseType(StatusCodes.Status404NotFound)]
+        [ProducesResponseType(StatusCodes.Status500InternalServerError)]
+        [SwaggerResponse(202, "Returns updated object containing user settings.")]
+        [SwaggerResponse(400, "The model state is invalid or there is a mismatch.")]
+        [SwaggerResponse(401, "An unauthorized request cannot be processed.")]
+        [SwaggerResponse(404, "No settings found by the provided id.")]
+        [SwaggerResponse(500, "An unhandled exception occurred on the server while executing the request.")]
+        public async Task<ActionResult<SettingsViewModel>> UpdateSettings(
+            Guid id,
+            [FromBody] SettingsViewModel settingsViewModel)
+        {
+            if (id != settingsViewModel.Id)
+            {
+                return this.BadRequest("SettingsId mismatch.");
+            }
+
+            var modelToUpdate = await this.settingsService.GetSettingsAsync(id);
+            if (modelToUpdate is null)
+            {
+                return this.NotFound("No settings found by the provided id.");
+            }
+
+            var model = this.mapper.Map<SettingsModel>(settingsViewModel);
+            model = await this.settingsService.UpdateSettingsAsync(model);
```

Comment:
We retrieve some data from from service, check it and then make decision to perform some action with same service. It's look like business logic in this case. Service became just a wrapper over repository.

Reply:
Got it, I'll move this logic to the service method.